File: ParkPal.Common/Models/Database/Entities/Notification/Subscription.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;
using ParkPal.Common.Models.Database.Entities.Device;

namespace ParkPal.Common.Models.Database.Entities.Notification
{
    [Table("Subscription", Schema = "Notification")]
    public class Subscription
    {
        [Key]
        public int SubscriptionId { get; set; }
        
        [ForeignKey("Token")]
        public int TokenId { get; set; }
        public Token? Token { get; set; }
        
        [MaxLength(1000)]
        public string PlayerId { get; set; }
        
        public virtual IEnumerable<AttractionTimer> AttractionTimers { get; set; }
    }
}

File: ParkPal.Application/src/models/api/Subscription.ts
import AttractionTimer from "@/models/api/AttractionTimer";

export default class Subscription {
    subscriptionId?: number;
    tokenId?: number;
    token?: Token;
    playerId?: string;
    attractionTimers: Array<AttractionTimer> = [];

    constructor(data: Pick<Subscription, "subscriptionId" | "tokenId" | "token" | "playerId" | "attractionTimers"> | null = null) {
        if(data != null) {
            this.subscriptionId = data.subscriptionId;
            this.tokenId = data.tokenId;
            this.token = data.token;
            this.playerId = data.playerId;

            if(data.attractionTimers) {
                data.attractionTimers.forEach((attractionTimer: AttractionTimer) => {
                    this.attractionTimers.push(attractionTimer);
                });
            }
        }
    }
}